nordicsemi/dfu/dfu_transport_serial.py

```diff
@@ -302,6 +304,38 @@ def try_to_recover():
 
             self._send_event(event_type=DfuEvent.PROGRESS_EVENT, progress=len(data))
 
+    def __ensure_bootloader(self):
+        lister = DeviceLister()
+        device = lister.get_device(com = self.com_port)
+        if device:
+            device_serial_number = device.serial_number
+
+            if not self.__is_device_in_bootloader_mode(device):
+                trigger = DFUTrigger()
+                trigger.enter_bootloader_mode(device)
+                logger.debug("Serial: DFU bootloader was triggered")
+
+                retry_count = 10
+                wait_time_ms = 500
+                for checks in range(retry_count):
+                    logger.debug("Serial: Waiting {} ms for device to enter bootloader {}/{} time".format(500, checks + 1, retry_count))
+                    time.sleep(wait_time_ms / 1000.0)
+
+                    device = lister.get_device(serial_number = device_serial_number)
+                    if self.__is_device_in_bootloader_mode(device):
+                        self.com_port = device.get_first_available_com_port()
+                        break
+
+                trigger.clean()
+            if not self.__is_device_in_bootloader_mode(device):
+                logger.debug("Serial: Device is either not in bootloader mode, or using an unsupported bootloader.")
+
+    def __is_device_in_bootloader_mode(self, device):
+        if not device:
+            return False
+        return device.vendor_id.lower() == '1915' and device.product_id.lower() == '521f'
```

Comment:
Could add a comment that explains why we are looking for this vid/pid combination